### PR TITLE
feat: bulk operations in resource browser (KubeCon 2024)

### PR DESCRIPTION
**Pull request by Elessar1802**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -155,7 +155,8 @@ if (!window || !window._env_) {
         FEATURE_HIDE_USER_DIRECT_PERMISSIONS_FOR_NON_SUPER_ADMINS: false,
         FEATURE_PROMO_EMBEDDED_BUTTON_TEXT: '',
         FEATURE_PROMO_EMBEDDED_MODAL_TITLE: '',
-        FEATURE_PROMO_EMBEDDED_IFRAME_URL:''
+        FEATURE_PROMO_EMBEDDED_IFRAME_URL:'',
+        FEATURE_BULK_RESTART_WORKLOADS_FROM_RB: 'deployMent,rOllOut,daemonset, statefulset'
```
Comment (vivek-devtron) [inline src/index.tsx:159]: the values should either be in small or caps

File: src/components/ClusterNodes/NodeDetailsList.tsx [hunk 2]
```diff
@@ -244,6 +280,21 @@ export default function NodeDetailsList({ isSuperAdmin, renderRefreshBar, addTab
         getNodeListData()
     }, [clusterId])
 
+    useEffect(() => {
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/NodeDetailsList.tsx:283]: It seems duplicated code. Please remove

File: src/components/ClusterNodes/NodeDetailsList.tsx [hunk 3]
```diff
@@ -362,18 +413,100 @@ export default function NodeDetailsList({ isSuperAdmin, renderRefreshBar, addTab
...
+            })
+        }
+    }
+
+    const renderNodeListHeader = (column: ColumnMetadataType): JSX.Element => {
+        if (column.label.toUpperCase() === 'NODE') {
+            return (
+                <div className="flexbox dc__gap-8 dc__align-items-center">
+                    <BulkSelection showPagination={showPaginatedView} />
+                    <SortableTableHeaderCell
```
Comment (vivek-devtron) [inline src/components/ClusterNodes/NodeDetailsList.tsx:455]: please check if we can common out this

File: src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx [hunk 4]
```diff
@@ -430,20 +563,39 @@ export const K8SResourceList = ({
...
-                        sortOrder={sortOrder}
-                        disabled={false}
-                    />
-                ))}
+                {resourceList?.headers.map((columnName, index) => {
+                    if (index === 0) {
+                        return (
+                            <div className="flexbox dc__gap-8 dc__align-items-center">
+                                <BulkSelection showPagination={showPaginatedView} />
+                                <SortableTableHeaderCell
```
Comment (vivek-devtron) [inline src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx:574]: extract out the common code

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts [hunk 5]
```diff
@@ -161,6 +161,7 @@ export const updateManifestResourceHelmApps = (
     updatedManifest: string,
     isResourceBrowserView?: boolean,
     selectedResource?: SelectedResourceType,
+    signal?: AbortSignal,
```
[the remark this answers is not in the document]
Comment (Elessar1802) [reply src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts:164]: this is required. I forgot to integrate with the api call.

**Pull request merged**